ddtrace/ext/ci.py

```diff
@@ -41,16 +44,28 @@
 
 
 def _normalize_ref(name):
+    # type: (Optional[str]) -> Optional[str]
     return _RE_TAGS.sub("", _RE_ORIGIN.sub("", _RE_REFS.sub("", name))) if name is not None else None
 
 
 def _filter_sensitive_info(url):
+    # type: (Optional[str]) -> Optional[str]
     return _RE_URL.sub("\\1", url) if url is not None else None
 
 
+def resolve_relative_workspace_path(filepath, env):
+    # type: (str, MutableMapping[str, str]) -> str
+    """Expand ~ in path to include home directory."""
+    if filepath.startswith("~") and (len(filepath) == 1 or filepath[1] == "/"):
```

Comment:
What about https://docs.python.org/3/library/os.path.html#os.path.expanduser ?

Reply:
Oh, it looks like this was using this before, why did you need to change it?

Reply:
I tried out expanduser while I was testing locally and it didn't work 😆 I'm pretty sure it works most of the time, but this was just to be sure it never gets broken

Reply:
seems like this implementation might not work for `~user/`?

Reply:
What situation would this occur? Don't these ~ expressions have to be followed by `/`?

Reply:
Update: ended up reverting back to `expanduser()` 😄